Library/Homebrew/cask/cask.rb

```diff
@@ -140,21 +157,21 @@ def outdated_versions(greedy: false, greedy_latest: false, greedy_auto_updates:
       # special case: tap version is not available
       return [] if version.nil?
 
-      if greedy || (greedy_latest && greedy_auto_updates) || (greedy_auto_updates && auto_updates)
-        return versions if version.latest?
-      elsif greedy_latest && version.latest?
-        return versions
-      elsif auto_updates
-        return []
-      end
-
       latest_version = if Homebrew::EnvConfig.install_from_api? &&
                           (latest_cask_version = Homebrew::API::Versions.latest_cask_version(token))
         DSL::Version.new latest_cask_version.to_s
       else
         version
       end
 
+      if greedy || greedy_latest || (greedy_auto_updates && auto_updates)
+        if latest_version.latest?
+          return outdated_download_sha? ? versions : []
+        end
+      elsif auto_updates
+        return []
+      end
```

Comment:
```suggestion
      if greedy || greedy_latest || (greedy_auto_updates && auto_updates)
        if latest_version.latest?
          if outdated_download_sha?
            return versions
          else
            return []
          end
        end
      elsif auto_updates
        return []
      end
```

Reply:
Unfortunately, this doesn't pass the linter. We could use a guard clause but that seems even uglier than the ternary operator imo. Which do you prefer?

```ruby
# before
return outdated_download_sha? ? versions : []

# after
return versions if outdated_download_sha?

return []
```

Reply:
# after
return versions if outdated_download_sha?
return []

This seems nice!